crates/matrix-sdk/src/test_utils/events.rs

```diff
@@ -238,6 +250,72 @@ impl EventFactory {
         builder
     }
 
+    /// Create a replacement message for an existing one given its id.
+    pub fn replacement_msg(
+        &self,
+        new_content: impl Into<String>,
+        original_event_id: &EventId,
+    ) -> EventBuilder<RoomMessageEventContent> {
+        let msg_content = RoomMessageEventContent::text_plain(new_content.into());
+        let builder = self.event(msg_content.clone());
+        builder.edit(original_event_id, msg_content.into())
+    }
+
+    /// Create a poll start event given a text, the question and the possible
+    /// answers.
+    pub fn poll_start(
+        &self,
+        content: impl Into<String>,
+        poll_question: impl Into<String>,
+        answers: Vec<impl Into<String>>,
+    ) -> EventBuilder<PollStartEventContent> {
+        // PollAnswers 'constructor' is not public, so we need to deserialize them
```

Comment:
It seems you can convert a `Vec<PollAnswer>` `into()` a `PollAnswers`; can you do this instead of the json hack please?

Reply:
```
error[E0277]: the trait bound `PollAnswers: From<Vec<PollAnswer>>` is not satisfied
   --> crates/matrix-sdk/src/test_utils/events.rs:267:32
    |
267 |         }).collect::<Vec<_>>().into();
    |                                ^^^^ the trait `From<Vec<PollAnswer>>` is not implemented for `PollAnswers`, which is required by `Vec<PollAnswer>: Into<_>`
    |
    = note: required for `Vec<PollAnswer>` to implement `Into<PollAnswers>`
```

Reply:
So it seems like this was fixed in Ruma, but our fork doesn't contain the change 🫤 .

Reply:
Ok, I just saw it does implement `TryFrom` in the commit hash we use 🤦 . Fixed in https://github.com/matrix-org/matrix-rust-sdk/pull/3870/commits/cc3c20ed27e08d3c0209e073d20865e25455b2fa.